### PR TITLE
docs: Steps for migrating to Envoy Gateway

### LINKED ISSUE #3742: docs: Steps for migrating to Envoy Gateway
*Description*:

Add a doc that can help users who want to migrate to Envoy Gateway

Would be great to highlight the ingress2gateway https://github.com/kubernetes-sigs/ingress2gateway/tree/main tool with an example

### PR DESCRIPTION
**Pull request by Manoramsharma**
**What type of PR is this?**
docs: Add a doc that can help users who want to migrate to Envoy Gateway

**What this PR does / why we need it**:
This PR highlights the `ingress2gateway` tool that can facilitate the migration to envoy gateway 

**Which issue(s) this PR fixes**:

Fixes #3742

### CODE REVIEW

File: site/content/en/latest/install/migrating-to-envoy.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
+---
+title: Migrating to Envoy Gateway
+---
+
+## Introduction
```
Comment (arkodg) [inline site/content/en/latest/install/migrating-to-envoy.md:5]: can we add a top level heading for `Migrating from Ingress Resources`

File: site/content/en/latest/install/migrating-to-envoy.md [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+2. **Kubernetes Cluster Access**: Ensure you have access to your Kubernetes cluster and necessary permissions to manage resources.
+3. **Installation of `ingress2gateway` Tool**: You need to install the `ingress2gateway` tool in your Kubernetes cluster and configure it accordingly. Follow the [ingress2gateway tool installation guide](https://github.com/kubernetes-sigs/ingress2gateway/blob/main/README.md#installation) for details.
+
+## Example Migration
+
+Here’s a step-by-step example of migrating from Ingress to Envoy Gateway using `ingress2gateway`:
+
+1. **Convert Ingress Resources**:
+
+   Run the following command to convert your Ingress resources into Gateway API resources:
```
Comment (arkodg) [inline site/content/en/latest/install/migrating-to-envoy.md:23]: can we add an example Ingress config here

**Review by arkodg — DISMISSED** [review]
thanks !
this is a good start, we can further enhance this doc in the future

**Review by Xunzhuo — COMMENTED** [review]
Should we mention that we need to configure the gatewayclass?

**Comment by Manoramsharma** [comment]
> Should we mention that we need to configure the gatewayclass?

Yes , we can . Could you please mention in which step, I will add this.

**Review by zirain — DISMISSED** [review]

**Comment by arkodg** [comment]
hey @Manoramsharma will the generated output be good enough or does the user need to also create a GatewayClass to link to the Envoy Gateway controller to ensure that the generated `HTTPRoutes`` get programmed in the data plane ?

**Comment by Manoramsharma** [comment]
> hey @Manoramsharma will the generated output be good enough or does the user need to also create a GatewayClass to link to the Envoy Gateway controller to ensure that the generated `HTTPRoutes`` get programmed in the data plane ?

Hi @arkodg you are right, GatewayClass defines the controller responsible for managing Gateways and programming the data plane. Without this linkage, the Gateway controller won't know which resources it should handle, and the generated HTTPRoutes won't be programmed correctly.

Let me just update the PR mentioning the steps to configure the Gateway class.

**Review by arkodg — APPROVED** [review]

**Review by zirain — APPROVED** [review]

**Pull request merged**